src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,119 @@
+package duke.utils;
+
+import java.time.LocalDateTime;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.CommandException;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.ToDoCommand;
+import duke.command.UnknownCommandException;
+import duke.command.UnmarkCommand;
+import duke.storage.LoadException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Class to parse information from one type to another.
+ * Types are bound to custom application types.
+ */
+public class Parser {
+    /**
+     * Returns a Command that corresponds to given input.
+     *
+     * @param input User input to application.
+     * @return Command to be executed.
+     * @throws CommandException Exception when input does not match with any known Command format.
+     */
+    public static Command parseInput(String input) throws CommandException {
+        input = input.stripLeading();
+
+        if (input.equals("bye")) {
+            return new ByeCommand();
+        } else if (input.equals("list")) {
+            return new ListCommand();
+        } else if (input.startsWith("todo")) {
+            return new ToDoCommand(input.substring(4));
+        } else if (input.startsWith("deadline")) {
+            return new DeadlineCommand(input.substring(8));
+        } else if (input.startsWith("event")) {
+            return new EventCommand(input.substring(5));
+        } else if (input.startsWith("mark")) {
+            return new MarkCommand(input.substring(4));
+        } else if (input.startsWith("unmark")) {
+            return new UnmarkCommand(input.substring(6));
+        } else if (input.startsWith("delete")) {
+            return new DeleteCommand(input.substring(6));
+        } else if (input.startsWith("find ")) {
+            // Account for space since parameters will not be trimmed.
+            return new FindCommand(input.substring(5));
+        }
```

Comment:
I like how you used PascalCase for class names, and that I can easily infer the purpose of the class based on the class names.